README.md

```diff
@@ -16,12 +16,17 @@ On the other hand, for certain bulk bit operations
 
 ## Thread safety
 
-* `Data.Bit` is faster, but writes and flips are thread-unsafe.
+* `Data.Bit` is faster, but writes and flips are not thread-safe.
   This is because naive updates are not atomic:
   they read the whole word from memory,
   then modify a bit, then write the whole word back.
+  Concurrently modifying separate slices of the same underlying array
```

Comment:
Maybe “adjacent” instead of “separate”?

Reply:
"Adjacent" implies that they're directly next to each other, which doesn't have to be the case. I first used "disjoint" (which would be the precise term), but figured that might be too jargony. I realize that "separate" can be misunderstood. Perhaps "non-intersecting"?

Reply:
“Non-intersecting” is good.